### DISCUSSION TITLE
Ft811 and esp32

### DISCUSSION THREAD

**Discussion by rezaaryi** [RudolphRiedel/FT800-FT813#104] [General]
Hello
I have more than a year
I bought FT811 and only succeeded with arduino atmega2560
LCD
7 inches
to launch
Of course with another library
ESP32 family library
does not support
Until I found your library
But I don't know platform io
An example image please
From esp32 and ft811 compilation
putting up
Please, because I don't understand at all
hamzehparsiniya@gmail.com

**Comment by rezaaryi** [comment]
innolux-LCD 7.0 inch AT070TN94
This is my lcd

**Comment by rezaaryi** [comment]
#
[FTDI_RNGO.ir.zip](https://github.com/RudolphRiedel/FT800-FT813/files/13852465/FTDI_RNGO.ir.zip)

**Comment by RudolphRiedel** [comment]
Hello,

my Arduino example compiles for a wide range of different Arduinos already:
https://github.com/RudolphRiedel/FT800-FT813/tree/5.x/examples/EVE_Test_Arduino_PlatformIO

The Platformio configuration is done with this file:
https://github.com/RudolphRiedel/FT800-FT813/blob/5.x/examples/EVE_Test_Arduino_PlatformIO/platformio.ini

First you have this:
build_flags = -D EVE_RVT50H

It selects the set of display parameters to be used from EVE_config.h.
Change it to:
build_flags = -D EVE_FT811CB_HY50HD

This might already be similar enough to directly work with your display, if not I would use set as a starting point.

Next is the board configuration in platformio.ini, you find something like this:
[env:ESP32]
platform = espressif32
board = esp32doit-devkit-v1
build_unflags = -Os
build_flags =
        ${env.build_flags}
;         -D EVE_USE_ESP_IDF
         -D EVE_CS=13
         -D EVE_PDN=12
         -D EVE_SCK=18
         -D EVE_MISO=19
         -D EVE_MOSI=23
         -O2

So the "board" is to tell PlatformIO which one you are using:
https://docs.platformio.org/en/latest/platforms/espressif32.html

And I noticed that this has
build_unflags = -Os
-O2
But I recently found that this is not a good option for plain ESP32 as the memory map that
PlatformIO is setting up is so tight that building with -O2 means that the bootloader is getting larger than the bootloader section.
I need to change that - and I probably need to go over all examples.

So selecting an Adafruit ItsyBitsy ESP32 and leaving the optimization to default looks like this:
[env:ESP32]
platform = espressif32
board = adafruit_itsybitsy_esp32
;build_unflags = -Os
build_flags =
        ${env.build_flags}
;         -D EVE_USE_ESP_IDF
         -D EVE_CS=13
         -D EVE_PDN=12
         -D EVE_SCK=18
         -D EVE_MISO=19
         -D EVE_MOSI=23
;         -O2

Now for the pin configuration, just to make this complete, you probably already figured this out:
EVE_SCK=18

So this sets the SPI SCK pin that the library uses to the Arduino pin 18.

The less obvious part is where this ends up.
In this case in https://github.com/RudolphRiedel/FT800-FT813/blob/5.x/EVE_target/EVE_target_Arduino_ESP32.h
These macros get defined between line 61 and 81 if they are not already defined.
And then these are used for the inline functions.

**Comment by rezaaryi** [comment]
Please install this library once as a movie
No matter how hard I try, I can't. I've been begging you for a year, but you didn't do it
hamzehparsiniya@gmail.com

**Comment by rezaaryi** [comment]
ft8111  and esp32-s3

**Reply by RudolphRiedel** [reply]
Ok, what can I do to help more?
What software development environment are you actually using and how is that not working?

I have been using an ESP32-S3 just yesterday and today I was using an ESP32-S2.
I was using ESP-IDF code with platformio though and I was using FT813, BT815 and BT817.
The FT811 is in no way special, it only uses less RGB lines and therefore supports fewer colors.

I do have a FT811CB from HAOYU here.
But I have not used it in years and it is hardwired to an AT90CAN128 - I just hooked it up to my bench PSU and it is not doing much.
No idea which software originally was on there.

Edit:
So, you have to work with me a little, I can try to answer your questions, I even can try to provide a working example project.
But I still need some details, what your issues are, which IDE you are using, which pins of your ESP32-S3 you are planning to use to connect to the FT811.

You are likely using the Arduino IDE, but there are 1.x and 2.x now.
Just a few weeks ago I setup a project using the Arduino IDE 2.x for an intern, running on the Arduino UNO R4.

And then there was this thread:
http://www.brtcommunity.com/index.php?topic=456.msg2285#msg2285

**Comment by rezaaryi** [comment]
Hello, thank you for reading my message
I do PCB design and I don't know programming with io platform, I don't know
how to import your library into io platform
There is a video on the internet about adding the library to the io
platform, but I can't understand it no matter what I look at
I do some programming work with Arduino
But I don't know how to import your library to the io platform with Visual
Studio and the io platform

در تاریخ پنجشنبه ۲۸ مارس ۲۰۲۴،‏ ۰:۴۵ RudolphRiedel ***@***.***>
نوشت:

**Comment by rezaaryi** [comment]
My problem is that I cannot add your library to the platformio

‫‪hamzeh parsiniya‬‏ ***@***.***‬‏> در تاریخ پنجشنبه ۲۸
مارس ۲۰۲۴ ساعت ۶:۴۰ نوشت:‬

**Comment by rezaaryi** [comment]
My problem is that I cannot add your library to the platform
That is, only you say it step by step or send me a completed project

‫‪hamzeh parsiniya‬‏ ***@***.***‬‏> در تاریخ پنجشنبه ۲۸
مارس ۲۰۲۴ ساعت ۶:۴۳ نوشت:‬

**Comment by rezaaryi** [comment]
I live in Iran
The biggest prison in the world
  My country is like a prison
Everything stops here
YouTube cut off Telegram, cut off WhatsApp and Facebook and many other
things
  There is a little bit of io platform tutorial
  I can't add your library

‫‪hamzeh parsiniya‬‏ ***@***.***‬‏> در تاریخ پنجشنبه ۲۸
مارس ۲۰۲۴ ساعت ۶:۴۹ نوشت:‬

**Comment by RudolphRiedel** [comment]
I am confused.
Do you actually want to use VSCode with PlatformIO and it just is not working for you?
Or do you want to use a different IDE? Then tell me which one, please.

**Comment by rezaaryi** [comment]
Yes, I use the io platform and Visual Studio, but it's basic and I don't
know how to add your library to the program. Would you like me to send you
a picture?

در تاریخ پنجشنبه ۲۸ مارس ۲۰۲۴،‏ ۱۲:۵۵ RudolphRiedel <
***@***.***> نوشت:

**Comment by RudolphRiedel** [comment]
Well, in this case it should be enough to open the folder https://github.com/RudolphRiedel/FT800-FT813/tree/5.x/examples/EVE_Test_ESP32_PlatformIO in VScode in order to at least compile this example.

The platformio.ini file in there has the project settings.

This is adding my library:
lib_deps = https://github.com/RudolphRiedel/FT800-FT813/archive/refs/heads/5.x.zip

And this selects the module to compile for:
build_flags = -D EVE_RVT50H

Change it to:
build_flags = -D EVE_FT811CB_HY50HD

But I already wrote this above.

Edit:
I tried it just now and changed to EVE_FT811CB_HY50HD, at least it compiled without issues for ESP32-S3.

**Comment by rezaaryi** [comment]
Hello, my dear friend, I am not dumb, I love Altium and board design,
welding inverter repairs, hospital equipment repairs, I spend all my time
learning these things.
Now I want to set up an esp32 board with ft811
And I don't know English. I have to translate all your words. The meaning
of the words you say may not be translated correctly. Please send me photos
of the steps.

در تاریخ پنجشنبه ۲۸ مارس ۲۰۲۴،‏ ۱۴:۵۱ RudolphRiedel <
***@***.***> نوشت:

**Reply by RudolphRiedel** [reply]
I never assumed you were dumb and English is not my first language either.
However, you refuse to answer my questions and do not give me much to work with.
You asked for images and I attach some, but I am pretty sure you do not actually need any of them.

Here, I modified my "EVE_Test_ESP32_PlatformIO" a bit, removed the other ESP32 variants and added EVE_FT811CB_HY50HD:
[EVE_Test_ESP32_S3_PlatformIO.zip](https://github.com/RudolphRiedel/FT800-FT813/files/14791421/EVE_Test_ESP32_S3_PlatformIO.zip)

Unpack the archive and open the folder in vscode:
![grafik](https://github.com/RudolphRiedel/FT800-FT813/assets/31180093/1524d761-a6c6-480d-9e45-fdde3ec87764)

Now build the project:
![grafik](https://github.com/RudolphRiedel/FT800-FT813/assets/31180093/34fd5b7c-4525-442b-b53c-889c2a1fb4a7)

If this worked, check the pin configuration in platformio.ini:
![grafik](https://github.com/RudolphRiedel/FT800-FT813/assets/31180093/3e579038-666e-4175-b0f2-9a46c8c33eac)

Adapt to the pins your hardware is actually using.
Now Upload this to your ESP32-S3 and report if this is working.

Please report any issue you encounter.
